lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua

```diff
@@ -91,20 +91,35 @@ function FuelPrices:AddPumpExtensions( pump )
     if SERVER then
         function pump:ChargeCustomer()
             local customer = pump.lastUser
-            local finalPrice = pump:CalculateFuelPrice()
+            local price = pump:CalculateFuelPrice()
 
-            if finalPrice == 0 then return end
+            if price == 0 then return end
 
             -- Ensures we don't charge them more than they have
             local customerMoney = customer:getDarkRPVar( "money" )
-            finalPrice = math.Clamp( finalPrice, 0, customerMoney )
+            price = math.Clamp( finalPrice, 0, customerMoney )
+
+            local priceStruct = {
+                price = price
+            }
+
+            function priceStruct:SetPrice( newPrice )
+                priceStruct.price = newPrice
+            end
+
+            local hookResult = hook.Run( "SimfPhys_FuelTaxes_WillChargeCustomer", pump, customer, priceStruct )
+            if hookResult == false then return end
+
+            local finalPrice = priceStruct.price
 
             local formattedPrice = DarkRP.formatMoney( math.Round( finalPrice, 2 ) )
 
             local message = "You've been charged " .. formattedPrice .. " for fuel"
             DarkRP.notify( customer, 0, 5, message)
 
             customer:addMoney( -finalPrice )
```

Comment:
Might wanna round money before taking from client? Not sure tho

Reply:
NO FREE FUEL

Reply:
To elaborate on my previous statement; the customer's wallet is stored as a float, so subtracting floats should work just fine, they just won't see it entirely accurate to the dollar (their display is rounded instead)

Reply:
If other addons also do this then I guess it's fine